### ISSUE TITLE
Plotting training loss across iterations

### ISSUE THREAD

**Issue by arunppsg** [deepchem/deepchem#3451]
I have a use case where I would like to plot training loss during training of a model in deepchem. Currently, to access the training loss for every iteration, I should enable logging in the script and parse the log file to get training log. Here is a sample code which does it:

```py
import logging
logging.basicConfig(filename='train.log', level=logging.INFO)

import deepchem as dc

tasks, datasets, transformers = dc.molnet.load_delaney(featurizer='GraphConv')
train_dataset, valid_dataset, test_dataset = datasets

model = dc.models.GraphConvModel(n_tasks=1, mode='regression', dropout=0.2, log_frequency=1)
```
This creates a file `train.log` which I need to parse further for accessing loss value. Is there a better way to do it?

**Comment by rbharath** [comment]
I think this is basically the best approach to pull out a full loss curve without having to use an external service like weights&biases
